qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -53,8 +53,15 @@ def __init__(self, basis=None, target=None):
         self._global_decomposers = None
         self._local_decomposers_cache = {}
 
-        if basis:
-            self._global_decomposers = _possible_decomposers(set(basis))
+        if self._basis_gates:
+            self._global_decomposers = _possible_decomposers(self._basis_gates)
```

Comment:
This loses the `set` creation (since it's not done on binding either).  Is that deliberate/important?

Reply:
Yeah, I moved it to the inside `_possible_decomposers` it was leftover from an earlier attempt I was making to simplify/deduplicate the logic between this and the no basis path that didn't work out. I'll revert this change and move it outside the function again.